forc-util/src/lib.rs

```diff
@@ -200,6 +201,49 @@ pub mod tx_utils {
     }
 }
 
+/// A `LogData` decoded into a human readable format with its type information.
+pub struct DecodedLog {
+    pub value: String,
+}
+
+pub fn decode_log_data(
+    log_id: &str,
+    log_data: &[u8],
+    program_abi: &ProgramABI,
+) -> anyhow::Result<DecodedLog> {
+    let program_abi = match program_abi {
+        ProgramABI::Fuel(fuel_abi) => Some(
+            fuel_abi_types::abi::unified_program::UnifiedProgramABI::from_counterpart(fuel_abi)?,
+        ),
+        _ => None,
+    }
+    .ok_or_else(|| anyhow::anyhow!("only fuelvm is supported for log decoding"))?;
+    // Create type lookup (id, TypeDeclaration)
+    let type_lookup = program_abi
+        .types
+        .iter()
+        .map(|decl| (decl.type_id, decl.clone()))
+        .collect::<HashMap<_, _>>();
+
+    let logged_type_lookup: HashMap<_, _> = program_abi
+        .logged_types
+        .iter()
+        .flatten()
+        .map(|logged_type| (logged_type.log_id.as_str(), logged_type.application.clone()))
+        .collect();
+
+    let type_application = logged_type_lookup
+        .get(&log_id)
+        .ok_or_else(|| anyhow::anyhow!("log id is missing"))?;
+
+    let abi_decoder = ABIDecoder::default();
+    let param_type = ParamType::try_from_type_application(type_application, &type_lookup)?;
+    let decoded_str = abi_decoder.decode_as_debug_str(&param_type, log_data)?;
+    let decoded_log = DecodedLog { value: decoded_str };
+
+    Ok(decoded_log)
+}
+
```

Comment:
I think this should be under `tx_utils` module so that it is feature gated:

https://github.com/FuelLabs/sway/blob/369eefd0d93302094879f779db83029cd04db24e/forc-util/src/lib.rs#L146-L202

Reply:
Otherwise you will be causing dependency on fuels in forc-pkg which in return creates a cyclic one and to prevent it we have a CI job which currently fails.

Reply:
addressed: [12f76b9](https://github.com/FuelLabs/sway/pull/6968/commits/12f76b9f590a8e2c7a8a4f07c7fcadb4a0c49a62)